src/System.Diagnostics.TraceSource/src/System/Diagnostics/DefaultTraceListener.cs

```diff
@@ -118,14 +111,12 @@ private void InitializeSettings()
 
         private void WriteAssert(string stackTrace, string message, string detailMessage)
         {
-            // Tracked by #32955: WriteAssert should indent "assertMessage" same way Debug.Fail does.
```

Comment:
Confirmed this was the correct behavior also seen in .NET Framework. So I updated Debug.Fail to match what we Write here.